addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/payloader/ExtensionPayloaderUnitTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrules.payloader;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.function.Supplier;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockedStatic;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.zap.extension.custompayloads.ExtensionCustomPayloads;
+import org.zaproxy.zap.extension.custompayloads.PayloadCategory;
+import org.zaproxy.zap.extension.pscanrules.ApplicationErrorScanRule;
+import org.zaproxy.zap.extension.pscanrules.ExtensionPscanRules;
+import org.zaproxy.zap.extension.pscanrules.UsernameIdorScanRule;
+import org.zaproxy.zap.testutils.TestUtils;
+
+class ExtensionPayloaderUnitTest extends TestUtils {
+
+    @Test
+    void shouldHaveAName() {
+        // Given
+        ExtensionPayloader ep = new ExtensionPayloader();
+        mockMessages(new ExtensionPscanRules());
+        // When / Then
+        assertThat(ep.getName(), is(equalTo("ExtensionPayloaderPscanRulesRelease")));
+        assertThat(ep.getUIName(), is(equalTo("Passive Scan Rules Release Custom Payloads")));
+    }
+
+    @Test
+    void shouldBeUnloadable() {
+        // Given
+        ExtensionPayloader ep = new ExtensionPayloader();
+        // When / Then
+        assertThat(ep.canUnload(), is(equalTo(true)));
+    }
+
+   @Test
```

Comment:
Should I:
```java
    @SuppressWarnings({
            "unchecked", "rawtypes"
    })
```

Or can we simply ignore it for UnitTests?

Reply:
Seems this was necessary: https://github.com/zaproxy/zap-extensions/actions/runs/5074594005/jobs/9114935451?pr=4611